src/data/list/zip.lean

```diff
@@ -200,14 +204,13 @@ theorem revzip_swap (l : list α) : (revzip l).map prod.swap = revzip l.reverse
 by simp [revzip]
 
 lemma nth_zip_with {α β γ} (f : α → β → γ) (l₁ : list α) (l₂ : list β) (i : ℕ) :
```

Comment:
Can you remove the local types now?
```suggestion
lemma nth_zip_with (f : α → β → γ) (l₁ : list α) (l₂ : list β) (i : ℕ) :
```
This type of thing is what causes accidental non-polymorphism in the first place.

Reply:
The universe monomorphism wasn't because of the local types. Rather, the general functor/applicative/monad notation is universe monomorphic because the general instances must be so. The local types were added to override the present variables.